AmplifyPlugins/Core/AWSPluginsCore/Model/Support/Model+GraphQL.swift

```diff
@@ -18,76 +18,129 @@ extension Model {
     func graphQLInput(_ modelSchema: ModelSchema) -> GraphQLInput {
         var input: GraphQLInput = [:]
         modelSchema.fields.forEach {
-            let field = $0.value
+            let modelField = $0.value
 
             // TODO how to handle associations of type "many" (i.e. cascade save)?
             // This is not supported right now and might be added as a future feature
-            if case .collection = field.type {
+            if case .collection = modelField.type {
                 return
             }
 
-            let name = field.graphQLName
-            let fieldOptionalValue: Any??
-
-            if let jsonModel = self as? JSONValueHolder {
-                fieldOptionalValue = jsonModel.jsonValue(for: field.name, modelSchema: modelSchema) ?? nil
-            } else {
-                fieldOptionalValue = self[field.name] ?? nil
-            }
+            let fieldValueOptional = getFieldValue(for: modelField.name, modelSchema: modelSchema)
 
             // Since the returned value is Any?? we need to do the following:
             // - `guard` to make sure the field name exists on the model
             // - `guard` to ensure the returned value isn't nil
-            guard let fieldValue = fieldOptionalValue else {
-                input.updateValue(nil, forKey: name)
+            guard let fieldValue = fieldValueOptional else {
+                input.updateValue(nil, forKey: modelField.name)
```

Comment:
We used `field.graphQLName` before? Just confirming both works?

Reply:
yes graphQLName returns name by default, and only in the case it's a belongsTo it will return `name.pascalCase()+"Id"`. i've refactored the logic down into the `case model` scenario

https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Core/AWSPluginsCore/Model/Support/ModelField+GraphQL.swift

Reply:
if it is belongsTo and returns targetName because targetName exists, then it may set targetName to nil here. I don't think this is ever the case because you cannot have nil belongsTo assocations, but i will revert this change and only make it if we're sure that there isn't a scenario that i've missed

Reply:
reverted this change